### DISCUSSION TITLE
Close Joyride programmatically

### DISCUSSION THREAD

**Discussion by mkantz84** [gilbarbara/react-joyride#626] [Q&A] [answered]
Hi,
I'm having problems navigating between pages while Joyride is running.
To solve the problems, I want to trigger `close` for the `tooltip`.
How can I do it?

**Comment by gilbarbara** [comment, marked as answer]
hey @mkantz84 

If you need to control how this library handles the tour, you'll need to use the controlled mode with `stepIndex` and `run` props
Check the [demo](https://github.com/gilbarbara/react-joyride-demo) for examples.

**Reply by mkantz84** [reply]
Thanks for the quick answer!
I'll try to explain more my problem.
I'm switching between 2 non controlled tour pages. But I do control the `run` prop, because I have a tour click in my app.
When switching I'm changing the `run` to `false`.
And now, in the second page, when clicking the tour, I'm changing `run` to `true`, the `index` (in `data` in `CallBackProps`) stays the same as before. Thus, it doesn't find the tour step.
If I'm closing the tour by pressing `close` in the first page, everything works perfectly.
That's why all I want is to trigger the `close` operation.
Is this even possible?
Am I doing something wrong?

Here's 2 videos showing working and non working cases:
[Captures.zip](https://github.com/gilbarbara/react-joyride/files/5785649/Captures.zip)

**Comment by gilbarbara** [comment]
Did you check the code for the controlled example in the demo?

It's impossible to know what's happening without seeing the code, but  I think it's possible to do with the `stepIndex`.
